<a href="https://colab.research.google.com/github/fabrizio-lanzetti/Analisis-/blob/main/eda_bank.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Acerca del conjunto de datos<br>
**Comercialización bancaria**<br>
*Resumen:*<br>
Los datos están relacionados con campañas de marketing directo (llamadas telefónicas) de una institución bancaria portuguesa. El objetivo de la clasificación es predecir si el cliente suscribirá un depósito a plazo (variable y).<br>

*Información del conjunto de datos:*<br>
Los datos están relacionados con campañas de marketing directo de una institución bancaria portuguesa. Las campañas de marketing se basaron en llamadas telefónicas. Muchas veces era necesario más de un contacto con un mismo cliente, para poder acceder si el producto (depósito a plazo bancario) estaría ('sí') o no ('no') suscrito.<br>

*Información de atributos:*<br>
**Datos del cliente bancario:**<br>
1. Edad (numérica)<br>
2. **Trabajo** : tipo de trabajo (categórico: 'administrador', 'obrero', 'emprendedor', 'criada', 'administración', 'jubilado', 'autónomo', 'servicios', 'estudiante', 'técnico', 'desempleado', 'desconocido')<br>
3. **Civil**: estado civil (categórico: 'divorciado', 'casado', 'soltero', 'desconocido'; nota: 'divorciado' significa divorciado o viudo)<br>
4. **Educación** (categórica: 'básico.4y', 'básico.6y', 'básico.9y', 'bachillerato', 'analfabeto', 'curso.profesional', 'título.universitario', 'desconocido')<br>
5. **Incumplimiento**: ¿tiene crédito en incumplimiento? (categórico: 'no', 'sí', 'desconocido')<br>
6. **Vivienda**: ¿tiene préstamo de vivienda? (categórico: 'no', 'sí', 'desconocido')<br>
7. **Préstamo**: ¿tiene préstamo personal? (categórico: 'no', 'sí', 'desconocido')<br>

**Relacionado con el último contacto de la campaña actual:**<br>
8. **Contacto**: tipo de comunicación del contacto (categórico: 'celular', 'teléfono')<br>
9. **Mes**: mes del año del último contacto (categórico: 'ene', 'feb', 'mar', ..., 'nov', 'dic')<br>
10. **Day_of_week**: último día de contacto de la semana (categórico: 'lunes', 'martes', 'miércoles', 'jueves', 'viernes')<br>
11. **Duración**: duración del último contacto, en segundos (numérico). Nota importante: este atributo afecta en gran medida el objetivo de salida (por ejemplo, si duración=0 entonces y='no'). Sin embargo, no se conoce la duración antes de realizar una llamada. Además, después del final de la llamada, obviamente se conoce y.<br>

**Otros atributos:**<br>
12. **Campaña**: número de contactos realizados durante esta campaña y para este cliente (numérico, incluye el último contacto)<br>
13. **Pdays**: número de días que pasaron después de que el cliente fue contactado por última vez desde una campaña anterior (numérico; 999 significa que el cliente no fue contactado previamente)<br>
14. **Anterior**: número de contactos realizados antes de esta campaña y para este cliente (numérico)<br>
15. **Resultado**: resultado de la campaña de marketing anterior (categórico: 'fracaso', 'inexistente', 'éxito')<br>

**Atributos del contexto social y económico:**<br>
16. **Emp.var.rate**: tasa de variación del empleo - indicador trimestral (numérico)<br>
17. **Cons.price.idx**: índice de precios al consumidor - indicador mensual (numérico)<br>
18. **Cons.conf.idx**: índice de confianza del consumidor - indicador mensual (numérico)<br>
19. **Euribor3m**: tipo de cambio euribor a 3 meses - indicador diario (numérico)<br>
20. **Nr.empleados**: número de empleados - indicador trimestral (numérico)<br>

**Variable de salida (objetivo deseado):**<br>
**y** - ¿el cliente ha suscrito un depósito a plazo? (binario: 'sí', 'no')<br>

*Pasos de análisis:*<br>
1. Análisis de información de atributos.
2. Aprendizaje automático (regresión logística, KNN, SVM, árbol de decisión, bosque aleatorio, Naive Bayes)
3. Aprendizaje profundo (ANN)


In [ ]:
import pandas as pd
import numpy as np
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots
from sklearn.preprocessing import OneHotEncoder

In [ ]:
#carga de archivos
bank = pd.read_csv("/content/drive/MyDrive/Proyecto Final DS /bank-additional-full.csv", sep=";")

In [ ]:
bank.head()

age        job  marital    education  default housing loan    contact  \
0   56  housemaid  married     basic.4y       no      no   no  telephone   
1   57   services  married  high.school  unknown      no   no  telephone   
2   37   services  married  high.school       no     yes   no  telephone   
3   40     admin.  married     basic.6y       no      no   no  telephone   
4   56   services  married  high.school       no      no  yes  telephone   

  month day_of_week  ...  campaign  pdays  previous     poutcome emp.var.rate  \
0   may         mon  ...         1    999         0  nonexistent          1.1   
1   may         mon  ...         1    999         0  nonexistent          1.1   
2   may         mon  ...         1    999         0  nonexistent          1.1   
3   may         mon  ...         1    999         0  nonexistent          1.1   
4   may         mon  ...         1    999         0  nonexistent          1.1   

   cons.price.idx  cons.conf.idx  euribor3m  nr.employed   y  
0          93.994          -36.4      4.857       5191.0  no  
1          93.994          -36.4      4.857       5191.0  no  
2          93.994          -36.4      4.857       5191.0  no  
3          93.994          -36.4      4.857       5191.0  no  
4          93.994          -36.4      4.857       5191.0  no  

[5 rows x 21 columns]

In [ ]:
bank.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41188 entries, 0 to 41187
Data columns (total 21 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   age             41188 non-null  int64  
 1   job             41188 non-null  object 
 2   marital         41188 non-null  object 
 3   education       41188 non-null  object 
 4   default         41188 non-null  object 
 5   housing         41188 non-null  object 
 6   loan            41188 non-null  object 
 7   contact         41188 non-null  object 
 8   month           41188 non-null  object 
 9   day_of_week     41188 non-null  object 
 10  duration        41188 non-null  int64  
 11  campaign        41188 non-null  int64  
 12  pdays           41188 non-null  int64  
 13  previous        41188 non-null  int64  
 14  poutcome        41188 non-null  object 
 15  emp.var.rate    41188 non-null  float64
 16  cons.price.idx  41188 non-null  float64
 17  cons.conf.idx   41188 non-null 

In [ ]:
bank.isnull().sum()

age               0
job               0
marital           0
education         0
default           0
housing           0
loan              0
contact           0
month             0
day_of_week       0
duration          0
campaign          0
pdays             0
previous          0
poutcome          0
emp.var.rate      0
cons.price.idx    0
cons.conf.idx     0
euribor3m         0
nr.employed       0
y                 0
dtype: int64

No tenemos datos nulos.

# EDA

In [ ]:
# EXPLORACION DE VARIABLES NUMERICAS
bank.describe()

age      duration      campaign         pdays      previous  \
count  41188.00000  41188.000000  41188.000000  41188.000000  41188.000000   
mean      40.02406    258.285010      2.567593    962.475454      0.172963   
std       10.42125    259.279249      2.770014    186.910907      0.494901   
min       17.00000      0.000000      1.000000      0.000000      0.000000   
25%       32.00000    102.000000      1.000000    999.000000      0.000000   
50%       38.00000    180.000000      2.000000    999.000000      0.000000   
75%       47.00000    319.000000      3.000000    999.000000      0.000000   
max       98.00000   4918.000000     56.000000    999.000000      7.000000   

       emp.var.rate  cons.price.idx  cons.conf.idx     euribor3m   nr.employed  
count  41188.000000    41188.000000   41188.000000  41188.000000  41188.000000  
mean       0.081886       93.575664     -40.502600      3.621291   5167.035911  
std        1.570960        0.578840       4.628198      1.734447     72.251528  
min       -3.400000       92.201000     -50.800000      0.634000   4963.600000  
25%       -1.800000       93.075000     -42.700000      1.344000   5099.100000  
50%        1.100000       93.749000     -41.800000      4.857000   5191.000000  
75%        1.400000       93.994000     -36.400000      4.961000   5228.100000  
max        1.400000       94.767000     -26.900000      5.045000   5228.100000

**Observaciones**<br>
**age**: Tiene un minimo de 17 años y un maximo de 98 años, su media y mediana no varian mucho, tienen una desviacion estandar de 10.42.<br>
**campaing**: Tiene un minimo de 1 y un maximo de 56, con una mediana ligeramente superior a la media y una desviacion estandar de 2.77.<br>
**previous**: Tiene un minimo de 0 y un maximo de 7, con una mediana de 0.17 y una media de 0, su desviacion es de 0.49.<br>
**emp.var.rate**: Tiene un minimo de -3.4 y una maxima de 1.4, con yna mediana de 0.8 y una media de 1.1, su desviacion es de 1.17.<br>
**cons.price.idx**: Tiene un minimo de 92 y un maximo de 94, su media y su mediana son casi iguales, con una desviacion estandar de 0.57, segun estas medidas de tendencias centrales podemos determinar una distribucion normal.<br>
**euribor3m**:Tiene una media de 3.621291, con una minima de 0.634000 y una maxima de 5.045000.<br>
**nr.employed**: Un promedio de 5167 empleados por entidad bancaria, con una minima de 4963 y una maxima de 5228.

In [ ]:
variables_del_grafico = ["age", "campaign", "previous", "emp.var.rate", "cons.price.idx", "euribor3m", "nr.employed"]
grafico = bank[variables_del_grafico]

def figuras(variable):
    fig = px.box(grafico, y=variable)
    fig.update_layout(
        template="plotly_dark",
        title=f"Boxplot de {variable.capitalize()}",
        yaxis_title=variable.capitalize(),
        showlegend=False,
        height=500,
        width=650
    )
    fig.show()

for variable in variables_del_grafico:
    figuras(variable)

 Observaciones
En los boxplots creados podemos ver un poco mas visual lo que se documento el DESCRIBE unas celdas antes. De esta manera nos vamos haciendo una idea visual de como se presentan los datos.

## EDA en variables categoricas.

In [ ]:
bank.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41188 entries, 0 to 41187
Data columns (total 21 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   age             41188 non-null  int64  
 1   job             41188 non-null  object 
 2   marital         41188 non-null  object 
 3   education       41188 non-null  object 
 4   default         41188 non-null  object 
 5   housing         41188 non-null  object 
 6   loan            41188 non-null  object 
 7   contact         41188 non-null  object 
 8   month           41188 non-null  object 
 9   day_of_week     41188 non-null  object 
 10  duration        41188 non-null  int64  
 11  campaign        41188 non-null  int64  
 12  pdays           41188 non-null  int64  
 13  previous        41188 non-null  int64  
 14  poutcome        41188 non-null  object 
 15  emp.var.rate    41188 non-null  float64
 16  cons.price.idx  41188 non-null  float64
 17  cons.conf.idx   41188 non-null 

### Vamos a analizar las variables pero separandolas en segmentos. Esto se hace con el fin de que el proyecto quede mas estructurado y entendible.

*  Primer segmento : trabajo, estado civil y educacuion
*  Segundo segmento: incumplimiento, vivienda, préstamo

*   Tercer segmento: contacto, pdays, resultado




### Primer segmento: Trabajo, estado civil y educacion.

In [ ]:
primer_segmento = ["job","marital","education"]
print(bank["job"].value_counts())
print("------------------------------")
print(bank["marital"].value_counts())
print("-------------------------------")
print(bank["education"].value_counts())

admin.           10422
blue-collar       9254
technician        6743
services          3969
management        2924
retired           1720
entrepreneur      1456
self-employed     1421
housemaid         1060
unemployed        1014
student            875
unknown            330
Name: job, dtype: int64
------------------------------
married     24928
single      11568
divorced     4612
unknown        80
Name: marital, dtype: int64
-------------------------------
university.degree      12168
high.school             9515
basic.9y                6045
professional.course     5243
basic.4y                4176
basic.6y                2292
unknown                 1731
illiterate                18
Name: education, dtype: int64


In [ ]:
fig = make_subplots(rows=1, cols=3, subplot_titles=['Trabajo', 'Estado civil', 'Educacuón'])

# Gráfico 1
default_counts = bank['job'].value_counts()
fig.add_trace(
    go.Bar(x=default_counts.index, y=default_counts.values),
    row=1, col=1
)

# Gráfico 2
housing_counts = bank['marital'].value_counts()
fig.add_trace(
    go.Bar(x=housing_counts.index, y=housing_counts.values),
    row=1, col=2
)

# Gráfico 3
loan_counts = bank['education'].value_counts()
fig.add_trace(
    go.Bar(x=loan_counts.index, y=loan_counts.values),
    row=1, col=3
)


fig.update_layout(
    template="plotly_dark",
    showlegend=False,
    height=400,
    width=1300,
)

fig.show()

### Segundo segmento: incumplimiento, vivienda, préstamo

In [ ]:
listtt = ["default","housing","loan"]
print(bank["default"].value_counts())
print("------------------------------")
print(bank["housing"].value_counts())
print("-------------------------------")
print(bank["loan"].value_counts())

no         32588
unknown     8597
yes            3
Name: default, dtype: int64
------------------------------
yes        21576
no         18622
unknown      990
Name: housing, dtype: int64
-------------------------------
no         33950
yes         6248
unknown      990
Name: loan, dtype: int64


In [ ]:
fig = make_subplots(rows=1, cols=3, subplot_titles=['Incumplimiento', 'Vivienda', 'Préstamo'])

# Gráfico 1
default_counts = bank['default'].value_counts()
fig.add_trace(
    go.Bar(x=default_counts.index, y=default_counts.values),
    row=1, col=1
)

# Gráfico 2
housing_counts = bank['housing'].value_counts()
fig.add_trace(
    go.Bar(x=housing_counts.index, y=housing_counts.values),
    row=1, col=2
)

# Gráfico 3
loan_counts = bank['loan'].value_counts()
fig.add_trace(
    go.Bar(x=loan_counts.index, y=loan_counts.values),
    row=1, col=3
)


fig.update_layout(
    template="plotly_dark",
    showlegend=False,
    height=400,
    width=1200,
)

fig.show()

### Tercer segmento: contacto, resultado, pdays

In [ ]:
fig = make_subplots(rows=1, cols=2, subplot_titles=['Contacto','Resultado'])

# Gráfico 1
default_counts = bank['contact'].value_counts()
fig.add_trace(
    go.Bar(x=default_counts.index, y=default_counts.values),
    row=1, col=1
)


# Gráfico 3
loan_counts = bank['poutcome'].value_counts()
fig.add_trace(
    go.Bar(x=loan_counts.index, y=loan_counts.values),
    row=1, col=2
)


fig.update_layout(
    template="plotly_dark",
    showlegend=False,
    height=400,
    width=1200,
)

fig.show()

# Data Transform
Lo primero que vamos a hacer es eliminar columnas que no usarmos. Luego, vamos a convertir todas nuestras columnas a tipo numericas, por ultimo, le aplicaremos one hot encoder.


In [ ]:
bank.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41188 entries, 0 to 41187
Data columns (total 21 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   age             41188 non-null  int64  
 1   job             41188 non-null  object 
 2   marital         41188 non-null  object 
 3   education       41188 non-null  object 
 4   default         41188 non-null  object 
 5   housing         41188 non-null  object 
 6   loan            41188 non-null  object 
 7   contact         41188 non-null  object 
 8   month           41188 non-null  object 
 9   day_of_week     41188 non-null  object 
 10  duration        41188 non-null  int64  
 11  campaign        41188 non-null  int64  
 12  pdays           41188 non-null  int64  
 13  previous        41188 non-null  int64  
 14  poutcome        41188 non-null  object 
 15  emp.var.rate    41188 non-null  float64
 16  cons.price.idx  41188 non-null  float64
 17  cons.conf.idx   41188 non-null 

In [ ]:
#eliminamos la variable duration
del bank["duration"] #usamos este metodo para que se elimine directamente del df

In [ ]:
#concatenamos las variables month y day_of_week, luego la pasamos a formato fecha y por ultimo borramos el año porque no lo tenemos y por defecto va a ser 1990
variables_fecha = ["month","day_of_week"]
bank['fecha'] = bank['month'] + ' ' + bank['day_of_week']

In [ ]:
bank['fecha'] = pd.to_datetime(bank['fecha'], format='%b %a')
bank.head(10)

age          job  marital            education  default housing loan  \
0   56    housemaid  married             basic.4y       no      no   no   
1   57     services  married          high.school  unknown      no   no   
2   37     services  married          high.school       no     yes   no   
3   40       admin.  married             basic.6y       no      no   no   
4   56     services  married          high.school       no      no  yes   
5   45     services  married             basic.9y  unknown      no   no   
6   59       admin.  married  professional.course       no      no   no   
7   41  blue-collar  married              unknown  unknown      no   no   
8   24   technician   single  professional.course       no     yes   no   
9   25     services   single          high.school       no     yes   no   

     contact month day_of_week  ...  pdays  previous     poutcome  \
0  telephone   may         mon  ...    999         0  nonexistent   
1  telephone   may         mon  ...    999         0  nonexistent   
2  telephone   may         mon  ...    999         0  nonexistent   
3  telephone   may         mon  ...    999         0  nonexistent   
4  telephone   may         mon  ...    999         0  nonexistent   
5  telephone   may         mon  ...    999         0  nonexistent   
6  telephone   may         mon  ...    999         0  nonexistent   
7  telephone   may         mon  ...    999         0  nonexistent   
8  telephone   may         mon  ...    999         0  nonexistent   
9  telephone   may         mon  ...    999         0  nonexistent   

  emp.var.rate  cons.price.idx  cons.conf.idx  euribor3m  nr.employed   y  \
0          1.1          93.994          -36.4      4.857       5191.0  no   
1          1.1          93.994          -36.4      4.857       5191.0  no   
2          1.1          93.994          -36.4      4.857       5191.0  no   
3          1.1          93.994          -36.4      4.857       5191.0  no   
4          1.1          93.994          -36.4      4.857       5191.0  no   
5          1.1          93.994          -36.4      4.857       5191.0  no   
6          1.1          93.994          -36.4      4.857       5191.0  no   
7          1.1          93.994          -36.4      4.857       5191.0  no   
8          1.1          93.994          -36.4      4.857       5191.0  no   
9          1.1          93.994          -36.4      4.857       5191.0  no   

       fecha  
0 1900-05-01  
1 1900-05-01  
2 1900-05-01  
3 1900-05-01  
4 1900-05-01  
5 1900-05-01  
6 1900-05-01  
7 1900-05-01  
8 1900-05-01  
9 1900-05-01  

[10 rows x 21 columns]

In [ ]:
bank = bank.drop(['month', 'day_of_week'], axis=1)

In [ ]:
bank.head(5)

age        job  marital    education  default housing loan    contact  \
0   56  housemaid  married     basic.4y       no      no   no  telephone   
1   57   services  married  high.school  unknown      no   no  telephone   
2   37   services  married  high.school       no     yes   no  telephone   
3   40     admin.  married     basic.6y       no      no   no  telephone   
4   56   services  married  high.school       no      no  yes  telephone   

   campaign  pdays  previous     poutcome  emp.var.rate  cons.price.idx  \
0         1    999         0  nonexistent           1.1          93.994   
1         1    999         0  nonexistent           1.1          93.994   
2         1    999         0  nonexistent           1.1          93.994   
3         1    999         0  nonexistent           1.1          93.994   
4         1    999         0  nonexistent           1.1          93.994   

   cons.conf.idx  euribor3m  nr.employed   y      fecha  
0          -36.4      4.857       5191.0  no 1900-05-01  
1          -36.4      4.857       5191.0  no 1900-05-01  
2          -36.4      4.857       5191.0  no 1900-05-01  
3          -36.4      4.857       5191.0  no 1900-05-01  
4          -36.4      4.857       5191.0  no 1900-05-01

**PDAYS**: Vamos a crear una variable nueva para obtener si el cliente fue contactado antes o no, ya que tenemos que si indica 999 es porque no se contacto.

In [ ]:
bank['llamada_cliente'] = (bank['pdays'] != 999).astype(int)
bank['dias_desde_ultimo_contacto'] = bank['pdays'].replace(999, 0)

In [ ]:
print(bank["llamada_cliente"].value_counts())
print("---------------------")
print(bank["pdays"].value_counts().index)
print("---------------------")
print(bank["dias_desde_ultimo_contacto"].value_counts().index)

0    39673
1     1515
Name: llamada_cliente, dtype: int64
---------------------
Int64Index([999,   3,   6,   4,   9,   2,   7,  12,  10,   5,  13,  11,   1,
             15,  14,   8,   0,  16,  17,  18,  22,  19,  21,  25,  26,  27,
             20],
           dtype='int64')
---------------------
Int64Index([ 0,  3,  6,  4,  9,  2,  7, 12, 10,  5, 13, 11,  1, 15, 14,  8, 16,
            17, 18, 22, 19, 21, 25, 26, 27, 20],
           dtype='int64')


In [ ]:
#eliminamos la variable pdays porque ya no la usamos
bank = bank.drop(["pdays"], axis=1)

In [ ]:
bank.head(5)

age        job  marital    education  default housing loan    contact  \
0   56  housemaid  married     basic.4y       no      no   no  telephone   
1   57   services  married  high.school  unknown      no   no  telephone   
2   37   services  married  high.school       no     yes   no  telephone   
3   40     admin.  married     basic.6y       no      no   no  telephone   
4   56   services  married  high.school       no      no  yes  telephone   

   campaign  previous     poutcome  emp.var.rate  cons.price.idx  \
0         1         0  nonexistent           1.1          93.994   
1         1         0  nonexistent           1.1          93.994   
2         1         0  nonexistent           1.1          93.994   
3         1         0  nonexistent           1.1          93.994   
4         1         0  nonexistent           1.1          93.994   

   cons.conf.idx  euribor3m  nr.employed   y      fecha  llamada_cliente  \
0          -36.4      4.857       5191.0  no 1900-05-01                0   
1          -36.4      4.857       5191.0  no 1900-05-01                0   
2          -36.4      4.857       5191.0  no 1900-05-01                0   
3          -36.4      4.857       5191.0  no 1900-05-01                0   
4          -36.4      4.857       5191.0  no 1900-05-01                0   

   dias_desde_ultimo_contacto  
0                           0  
1                           0  
2                           0  
3                           0  
4                           0

In [ ]:
lista_one_hot = ["job","marital","education","default","housing","loan","contact","poutcome","y"]

Las variables restantes vamos a mapearlas y cambiar por datos numericos para evitar crear tantas columnas.

In [ ]:
bank = pd.get_dummies(bank, columns=['job', 'marital', 'education', 'default', 'housing', 'loan', 'contact', 'poutcome'])


In [ ]:
bank.isnull().sum()

age                              0
campaign                         0
previous                         0
emp.var.rate                     0
cons.price.idx                   0
cons.conf.idx                    0
euribor3m                        0
nr.employed                      0
y                                0
fecha                            0
llamada_cliente                  0
dias_desde_ultimo_contacto       0
job_admin.                       0
job_blue-collar                  0
job_entrepreneur                 0
job_housemaid                    0
job_management                   0
job_retired                      0
job_self-employed                0
job_services                     0
job_student                      0
job_technician                   0
job_unemployed                   0
job_unknown                      0
marital_divorced                 0
marital_married                  0
marital_single                   0
marital_unknown                  0
education_basic.4y  

In [ ]:
bank.head(5)

age  campaign  previous  emp.var.rate  cons.price.idx  cons.conf.idx  \
0   56         1         0           1.1          93.994          -36.4   
1   57         1         0           1.1          93.994          -36.4   
2   37         1         0           1.1          93.994          -36.4   
3   40         1         0           1.1          93.994          -36.4   
4   56         1         0           1.1          93.994          -36.4   

   euribor3m  nr.employed   y      fecha  ...  housing_unknown  housing_yes  \
0      4.857       5191.0  no 1900-05-01  ...                0            0   
1      4.857       5191.0  no 1900-05-01  ...                0            0   
2      4.857       5191.0  no 1900-05-01  ...                0            1   
3      4.857       5191.0  no 1900-05-01  ...                0            0   
4      4.857       5191.0  no 1900-05-01  ...                0            0   

   loan_no  loan_unknown  loan_yes  contact_cellular  contact_telephone  \
0        1             0         0                 0                  1   
1        1             0         0                 0                  1   
2        1             0         0                 0                  1   
3        1             0         0                 0                  1   
4        0             0         1                 0                  1   

   poutcome_failure  poutcome_nonexistent  poutcome_success  
0                 0                     1                 0  
1                 0                     1                 0  
2                 0                     1                 0  
3                 0                     1                 0  
4                 0                     1                 0  

[5 rows x 50 columns]

In [ ]:
bank.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41188 entries, 0 to 41187
Data columns (total 50 columns):
 #   Column                         Non-Null Count  Dtype         
---  ------                         --------------  -----         
 0   age                            41188 non-null  int64         
 1   campaign                       41188 non-null  int64         
 2   previous                       41188 non-null  int64         
 3   emp.var.rate                   41188 non-null  float64       
 4   cons.price.idx                 41188 non-null  float64       
 5   cons.conf.idx                  41188 non-null  float64       
 6   euribor3m                      41188 non-null  float64       
 7   nr.employed                    41188 non-null  float64       
 8   y                              41188 non-null  object        
 9   fecha                          41188 non-null  datetime64[ns]
 10  llamada_cliente                41188 non-null  int64         
 11  dias_desde_ulti